doc/build/kconfig/extensions.rst

```diff
@@ -126,3 +126,35 @@ which includes some Kconfig extensions:
 - ``def_int``, ``def_hex``, and ``def_string`` keywords are available,
   analogous to ``def_bool``. These set the type and add a ``default`` at the
   same time.
+
+- A symbol name can be associated to ``choice`` groups using the ``choice <symbol>`` syntax.
+  Such choices, called *named choices*, can be modified by other places after their initial
+  definition. For example, if ``choice BOOT_SIGNATURE_TYPE`` was initially defined as:
+
+  .. code-block:: kconfig
+
+    choice BOOT_SIGNATURE_TYPE
```

Comment:
I think we can present this using `FOO` and `BAR` as we do with other examples.

Like:
```
choice FOO_CHOICE
    default FOO

    config FOO
	    bool "Foo"

    config BAR
	    bool "Bar"

endchoice
```

and then describe that the default can be changed.